pages/index.vue

```diff
@@ -117,7 +120,7 @@ watch(
               :track="item"
               :is-track-type-known="true"
               show-thumbnail
-              @play-track="setCurrentTrack(item)"
+              @play-track="() => { const items = group.items.filter((c) : c is TrackModel => c.type === 'track'); setQueue(items, items.findIndex(track => track.id === item.id)) }"
```

Comment:
Should we extract this to its own function? An inline function this large is not particularly readable in my opinion